src/Mono.Android/Test/Mono.Android-Tests.projitems

```diff
@@ -3,13 +3,18 @@
   <ItemGroup>
     <TestApk Include="$(OutputPath)Mono.Android_Tests-Signed.apk">
       <Package>Mono.Android_Tests</Package>
-      <InstrumentationType>xamarin.android.runtimetests.TestInstrumentation</InstrumentationType>
-      <ResultsPath>$(OutputPath)TestResult-Mono.Android_Tests.xml</ResultsPath>
       <TimingDefinitionsFilename>$(MSBuildThisFileDirectory)..\..\..\build-tools\scripts\TimingDefinitions.txt</TimingDefinitionsFilename>
       <TimingResultsFilename>$(MSBuildThisFileDirectory)..\..\..\TestResult-Mono.Android_Tests-times.csv</TimingResultsFilename>
       <ApkSizesInputFilename>apk-sizes-$(_MonoAndroidTestPackage)-$(Configuration)$(TestsAotName).txt</ApkSizesInputFilename>
       <ApkSizesDefinitionFilename>$(MSBuildThisFileDirectory)apk-sizes-definitions.txt</ApkSizesDefinitionFilename>
       <ApkSizesResultsFilename>$(MSBuildThisFileDirectory)..\..\..\TestResult-Mono.Android_Tests-values.csv</ApkSizesResultsFilename>
     </TestApk>
   </ItemGroup>
+
+  <ItemGroup>
+    <TestApkInstrumentation Include="xamarin.android.runtimetests.TestInstrumentation">
+      <Package>Mono.Android_Tests</Package>
+      <ResultsPath>$(OutputPath)TestResult-Mono.Android_Tests.xml</ResultsPath>
```

Comment:
Do we need `Package` here? It looks like it is duplicated in all the cases.

Reply:
I see it only used in the https://github.com/xamarin/xamarin-android/pull/1075/files#diff-6969d1e0d6b6392b34ee6cdcf770c84f and the Package is the same.

Wouldn't it be at least nice to have the `Package` set in a property, so that it can be set at single place?

Like:
```
<ItemGroup>
     <TestApk Include="$(OutputPath)Mono.Android_Tests-Signed.apk">
       <Package>$RuntimeTestPackage</Package>
...
<ItemGroup>
  <TestApkInstrumentation Include="xamarin.android.runtimetests.TestInstrumentation">
    <Package>$(RuntimeTestPackage)</Package>
    <ResultsPath>$(OutputPath)TestResult-Mono.Android_Tests.xml</ResultsPath>
...
```

Reply:
Package name in property name would be nice but it doesn't work in xbuild - been there done that :) It is also used [here](https://github.com/xamarin/xamarin-android/pull/1075/files#diff-30bb5e40ceb1a53bcbc0fa8d6d79b8bb). It's the only sensible mechanism to support multiple instrumentations per package. We can't put the information in `TestApk` element and there's also no sane way to "link" `<TestApkInstrumetation>` to a particular `TestApk` instance.

Reply:
I don't know why would it not work on xbuild. We did it before and it worked, like here: https://github.com/grendello/xamarin-android/blob/385699a58635d87a1cc65b60995d3420995f21d7/src/Mono.Android/Test/Mono.Android-Tests.projitems

Reply:
I tried that approach and, unfortunately, this is still broken. Using a property instead of the string makes `UndeployApkTests` target fail to process half of them when batching - this is the issue I had sometime ago while working on mono runtime builds. So, as much as it would be a nice change, we can't do it right now :(

Reply:
That is weird, it worked not so long time ago. What error do you get?

Reply:
No error at all... Half of the items using that property are simply ignored (in my case in `UndeployApks` and `DeployApks` which iterate over `@(TestApk)`) - this is the behavior I saw with the runtime stuff. It appears that sometimes xbuild fails to process properties inside item metadata.